### ISSUE TITLE
Getting encoding mode (--o=...) to be more robust and less prone to fatal-erroring like in GUI mode

### ISSUE THREAD

**Issue by hedgehog90** [mpv-player/mpv#8546]
I've been using MPV to transcode dynamic playlists, streaming the resulting output directly to an rtmp streaming server.
I need MPV to act exactly like it does in GUI mode - if I seek the currently playing file, this needs to be reflected in the output, and the same goes with if I enable or disable subtitles, change the volume, etc.
The desired result is to create an output stream that perfectly resembles the output if I was running it in normal GUI mode.

But currently there are some issues, particularly when it encounters any 'badly' encoded media file in the playlist. The following examples result in either mpv quitting or ending playback in encoding mode, but in GUI mode are handled much better:

If the player encounters bad audio PTS, Invalid video timestamps or non-monotonic PTS.
- **GUI Mode**: Attempts to fix the timings, usually does a brilliant job.
- **Encoding Mode**: fails within a second with 'Writing packet failed.' by the encoder)
  Can be improved by using --no-correct-pts and explicitly setting the --fps for each media file on_load, however some particularly badly encoded media I've come across (often the output from VLC's encoder if the metadata is to be trusted) will play but will be terribly out of sync.

If the player encounters corrupted segments in a media file
- **GUI Mode**: Skips to the first readable segment after the corrupted part.
- **Encoding Mode**: MPV doesn't exit but playback core is paused (core-idle=true), only way to restart is to reload or load another media file.

I mentioned it in the first example, but audio sync appears to be a big issue in encoding mode when it isn't in GUI mode.

It doesn't matter what encoder or demuxer I use, they always run into these same issues.
It seems the internal decoder when outputting to GUI is just much more robust than the one used in encoding mode, and I'm not sure why.

Any help towards improving this would be greatly appreciated, either through suggested source code hacks or command line options.

**Comment by Akemi** [comment]
since there hasn't been a reply till now, i am closing this.

**Comment by hedgehog90** [comment]
Since submitting this issue I implemented some improvements on my fork & haven't had a fatal encoding error since, but who knows if I haven't broken some functionality elsewhere as I'm quite unfamiliar with the encoder code.

I added a new option --ocontinue-on-fail (terribly named) so if the eof reason was due to an error it wouldn't skip every subsequent item in the playlist (wasn't sure if this was a feature or a bug tbh), but instead move onto the next and continue.
Also added a few lines to prevent PTS going backwards under any circumstances, this helped encoding some videos encoded with VLC that was prone to fucking up PTS values (imagine the PTS value linearly incrementing and then adding +/- noise).
The final improvement was to handle videos with missing or corrupted segments a bit better, so instead of quitting it just continued from the next playable point in the file. This is the least tested and hackiest 'improvement' though.

Let me know if this sounds like something worth adding to the main branch.
